VideoWeb/VideoWeb/ClientApp/src/app/services/unload-detector.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable, Renderer2, RendererFactory2 } from '@angular/core';
+import { DeviceDetectorService } from 'ngx-device-detector';
+import { Observable } from 'rxjs';
+import { Subject } from 'rxjs';
+import { filter, tap } from 'rxjs/operators';
+import { Logger } from './logging/logger-base';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class UnloadDetectorService {
+    private loggerPrefix = '[UnloadDetectorService] -';
+    private visibilityChangeSubject = new Subject<boolean>();
+    private shouldUnloadSubject = new Subject<void>();
+    private beforeUnloadSubject = new Subject<void>();
+
+    private renderer: Renderer2;
+    isDesktop: boolean;
+
+    constructor(private deviceDetectorService: DeviceDetectorService, renderer2Factor: RendererFactory2, private logger: Logger) {
```

Comment:
very minor, but probably should name the property to renderer2Factory instead of renderer2Factor, I was wonderig if that was just a typo?

Reply:
Yeah it was just a typo. Will fix now cheers.